src/components/coretemp/linux-coretemp.c

```diff
@@ -455,18 +455,18 @@ _coretemp_init_component( int cidx )
      }
 
      do {
-        char *strCpy;
-        strCpy=strncpy(_coretemp_native_events[i].name,t->name,PAPI_MAX_STR_LEN);
-        if (strCpy == NULL) HANDLE_STRING_ERROR;
+        int retlen;
+        retlen = snprintf(_coretemp_native_events[i].name, PAPI_MAX_STR_LEN, "%s", t->name);
+        if (retlen <= 0 || retlen >= PAPI_MAX_STR_LEN) HANDLE_STRING_ERROR;
```

Comment:
We should eventually replace `HANDLE_STRING_ERROR` with proper error handling. I think this patch looks good for the time being.

Reply:
There is a lot of existing code that uses `HANDLE_STRING_ERROR` and very few places using `PAPI_EBUF`. Switching to `PAPI_EBUF` does sound like a good change for a future patch set.

Reply:
Ok, I think you have a point. Let's keep this for consistency's sake.